doc/source/heterogeneous.rst

```diff
@@ -0,0 +1,222 @@
+
+   THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS" AND
+   ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE IMPLIED
+   WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+   DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY
+   DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES
+   (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES;
+   LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND
+   ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+   (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF THIS
+   SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+
+.. _heterogeneous:
+
+Heterogeneous Morphologies
+**************************
+
+.. image:: images/heterogeneous_neuron.png
+
+Definition
+----------
+
+A heterogeneous morphology consists of zero or more homogeneous and at least one heterogeneous neurite trees extending from the soma.
+A heterogeneous neurite tree consists of multiple sub-neurites with different types (ie: basal and axon).
+
+A typical example of a heterogeneous neurite is the axon-carrying dendrite, in which the axon sprouts from the basal dendrite.
+
+
+Identification
+--------------
+
+Heterogeneous neurites can be identified using the ``Neurite::is_heterogeneous`` method:
+
+.. code:: python
+
+    from neurom import load_morphology
+    from neurom.core.morphology import iter_neurites
+
+    m = load_morphology('tests/data/swc/heterogeneous_morphology.swc')
+
+    print([neurite.is_heterogeneous() for neurite in m])
+
+which would return ``[False, True, False]``, meaning the 2nd neurite extending from the soma contains multiple neurite types.
+
+
+Sub-neurite views of heterogeneous neurites
+--------------------------------------------
+
+Default mode
+~~~~~~~~~~~~
+
+NeuroM does not take into account heterogeneous sub-neurites by default.
+A heterogeneous neurite is treated as a homogeneous one, the type of which is determined by the first section of the tree.
+For example:
+
+.. code-block:: python
+
+    from neurom import load_morphology
+    from neurom.core.morphology import iter_neurites
+
+    m = load_morphology('tests/data/swc/heterogeneous_morphology.swc')
+
+    basal, axon_carrying_dendrite, apical = list(iter_neurites(m))
+
+    print(basal.type, axon_carrying_dendrite.type, apical.type)
+
+would print the types ``(basal_dendrite, basal_dendrite, apical_dendrite)``.
+I.E. the axon-carrying dendrite would be treated as a basal dendrite.
+For feature extraction and checks, the axon-carrying dendrite is treated as a basal dendrite.
+Features, for which an axon neurite type is passed, do not have access to the axonal part of the neurite.
+For instance, the number of basal and axon neurites will be two and zero respectively.
```

Comment:
Perhaps add what a feature would do in this case; ie, would the `volume` include the ACD or just the the basal portion?